commands2/proxycommand.py

```diff
@@ -43,6 +56,11 @@ def __init__(self, *args, **kwargs):
         def init_supplier(supplier: Callable[[], Command]):
             assert callable(supplier)
             self._supplier = supplier
+            warnings.warn(
+                "The ProxyCommand supplier constructor has been deprecated",
+                DeprecationWarning,
+                stacklevel=2,
```

Comment:
Ideally the warning should point to user code.

I haven't tested it, but I think since this is a nested function we want to go one step further up the stack.
```suggestion
                stacklevel=3,
```